### ISSUE TITLE
Issue with Stereographic Projection of a Polar Map

### ISSUE THREAD

**Issue by GiancorradoBrighi** [SciTools/cartopy#1924]
I am currently using cartopy V0.20.1, and I am trying to plot a Stereographic Map of the North Pole of Titan, the largest moon of Saturn. I downloaded a map of the North Pole of the planet from NASA's Photojournal website (https://photojournal.jpl.nasa.gov/beta/catalog/PIA17655), and as it can be seen the map goes from 50° to 90° in latitude. I wrote the following script:

```
def titan_image():

    raw_image = Image.open('/home/gbrighi/maps/NorthPole.png')

    # Resizing of the image
    basewidth = 500
    wpercent = (basewidth/float(raw_image.size[0]))
    hsize = int(float(raw_image.size[1])*float(wpercent))
    smaller_image = raw_image.resize((basewidth,hsize))
    img = np.asarray(smaller_image)

    titan_globe = ccrs.Globe(semimajor_axis = 2575.164*10**3, semiminor_axis = 2574.314*10**3, ellipse = None)
    img_proj    = ccrs.NorthPolarStereo(globe=titan_globe)
    img_extent  = (-180,180,50,90)
    return img, titan_globe, img_proj, img_extent

# Plot of the actual map:
fig = plt.figure(figsize=(6,6))
img, globe, crs, extent = titan_image()
my_proj = ccrs.NorthPolarStereo(globe = globe)
ax  = fig.add_subplot(1,1,1, projection = my_proj)
ax.set_extent([-180, 180, 50, 90], crs = ccrs.PlateCarree(globe = globe))
ax.gridlines(color = 'black', crs = ccrs.PlateCarree(globe = globe))
ax.imshow(img, transform=crs, extent = extent)

ax.plot(30,75, marker = 'o', color = 'r', label='Prova', transform = ccrs.PlateCarree(globe = globe))
plt.legend()
plt.show()
```

The output looks like a Polar Stereo map, limited from 50° to 90° in latitude, but the image is not shown in the background. The script does not produce any error, but the plot has a blank background with gridlines and the red dot plotted in the right position. 
The output image is provided at the following link: https://user-images.githubusercontent.com/93629139/140499209-8e79433b-27f2-4c61-9d1f-99eb2d85c578.png

Basically the problem lies in getting a Stereographic projection of a Stereographic map. If I use a similar script, but starting from a PlateCarree map of Titan, I obtain, as expected, a bounded stereo projection of it. 

```

### Cartopy version 0.20.1

**Comment by greglucas** [comment]
The image extent should be defined in the coordinates of the transform. So, your `imshow()` call should be something like:

```python
ax.imshow(img, transform=crs, extent=ax.get_extent())
```

since you've already set the extent of the image to use PlateCarree bounds, you'll see that the `ax.get_extent()` bounds are something huge like: `(-1866891.9293306577, 1856938.5729849285, -1842950.859294265, 1874895.347291459)`.

**Comment by GiancorradoBrighi** [comment]
Thank you very much. It **works**!
